File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientCpp/src/StateAlterationModelView.cpp
#include "../inc/StateAlterationModelView.h"
void StateAlterationModelView::serialize(std::ostream& output) {
	// Type
	output << ((int)this->Type) << '\n';
	// BaseDuration
	output << ((float)this->BaseDuration) << '\n';
	// DashGoThroughWall
	output << (this->DashGoThroughWall ? 1 : 0) << '\n';
	// DashDirectionType
	output << ((int)this->DashDirectionType) << '\n';
	// FlatValue
	output << ((float)this->FlatValue) << '\n';
	// SourcePercentADValue
	output << ((float)this->SourcePercentADValue) << '\n';
	// SourcePercentHPValue
	output << ((float)this->SourcePercentHPValue) << '\n';
	// SourcePercentMaxHPValue
	output << ((float)this->SourcePercentMaxHPValue) << '\n';
	// SourcePercentArmorValue
	output << ((float)this->SourcePercentArmorValue) << '\n';
	// SourcePercentAPValue
	output << ((float)this->SourcePercentAPValue) << '\n';
	// SourcePercentRMValue
	output << ((float)this->SourcePercentRMValue) << '\n';
	// DestPercentADValue
	output << ((float)this->DestPercentADValue) << '\n';
	// DestPercentHPValue
	output << ((float)this->DestPercentHPValue) << '\n';
	// DestPercentMaxHPValue
	output << ((float)this->DestPercentMaxHPValue) << '\n';
	// DestPercentArmorValue
	output << ((float)this->DestPercentArmorValue) << '\n';
	// DestPercentAPValue
	output << ((float)this->DestPercentAPValue) << '\n';
	// DestPercentRMValue
	output << ((float)this->DestPercentRMValue) << '\n';
	// StructureBonus
	output << ((float)this->StructureBonus) << '\n';
	// MonsterBonus
	output << ((float)this->MonsterBonus) << '\n';
	// CreepBonus
	output << ((float)this->CreepBonus) << '\n';
}

StateAlterationModelView StateAlterationModelView::deserialize(std::istream& input) {
	StateAlterationModelView _obj = StateAlterationModelView();
	// Type
	int _obj_Type; input >> _obj_Type; input.ignore(1000, '\n');
	_obj.Type = (::StateAlterationType)_obj_Type;
	// BaseDuration
	float _obj_BaseDuration; input >> _obj_BaseDuration; input.ignore(1000, '\n');
	_obj.BaseDuration = (float)_obj_BaseDuration;
	// DashGoThroughWall
	bool _obj_DashGoThroughWall; input >> _obj_DashGoThroughWall; input.ignore(1000, '\n');
	_obj.DashGoThroughWall = (bool)_obj_DashGoThroughWall;
	// DashDirectionType
	int _obj_DashDirectionType; input >> _obj_DashDirectionType; input.ignore(1000, '\n');
	_obj.DashDirectionType = (::DashDirectionType)_obj_DashDirectionType;
	// FlatValue
	float _obj_FlatValue; input >> _obj_FlatValue; input.ignore(1000, '\n');
	_obj.FlatValue = (float)_obj_FlatValue;
	// SourcePercentADValue
	float _obj_SourcePercentADValue; input >> _obj_SourcePercentADValue; input.ignore(1000, '\n');
	_obj.SourcePercentADValue = (float)_obj_SourcePercentADValue;
	// SourcePercentHPValue
	float _obj_SourcePercentHPValue; input >> _obj_SourcePercentHPValue; input.ignore(1000, '\n');
	_obj.SourcePercentHPValue = (float)_obj_SourcePercentHPValue;
	// SourcePercentMaxHPValue
	float _obj_SourcePercentMaxHPValue; input >> _obj_SourcePercentMaxHPValue; input.ignore(1000, '\n');
	_obj.SourcePercentMaxHPValue = (float)_obj_SourcePercentMaxHPValue;
	// SourcePercentArmorValue
	float _obj_SourcePercentArmorValue; input >> _obj_SourcePercentArmorValue; input.ignore(1000, '\n');
	_obj.SourcePercentArmorValue = (float)_obj_SourcePercentArmorValue;
	// SourcePercentAPValue
	float _obj_SourcePercentAPValue; input >> _obj_SourcePercentAPValue; input.ignore(1000, '\n');
	_obj.SourcePercentAPValue = (float)_obj_SourcePercentAPValue;
	// SourcePercentRMValue
	float _obj_SourcePercentRMValue; input >> _obj_SourcePercentRMValue; input.ignore(1000, '\n');
	_obj.SourcePercentRMValue = (float)_obj_SourcePercentRMValue;
	// DestPercentADValue
	float _obj_DestPercentADValue; input >> _obj_DestPercentADValue; input.ignore(1000, '\n');
	_obj.DestPercentADValue = (float)_obj_DestPercentADValue;
	// DestPercentHPValue
	float _obj_DestPercentHPValue; input >> _obj_DestPercentHPValue; input.ignore(1000, '\n');
	_obj.DestPercentHPValue = (float)_obj_DestPercentHPValue;
	// DestPercentMaxHPValue
	float _obj_DestPercentMaxHPValue; input >> _obj_DestPercentMaxHPValue; input.ignore(1000, '\n');
	_obj.DestPercentMaxHPValue = (float)_obj_DestPercentMaxHPValue;
	// DestPercentArmorValue
	float _obj_DestPercentArmorValue; input >> _obj_DestPercentArmorValue; input.ignore(1000, '\n');
	_obj.DestPercentArmorValue = (float)_obj_DestPercentArmorValue;
	// DestPercentAPValue
	float _obj_DestPercentAPValue; input >> _obj_DestPercentAPValue; input.ignore(1000, '\n');
	_obj.DestPercentAPValue = (float)_obj_DestPercentAPValue;
	// DestPercentRMValue
	float _obj_DestPercentRMValue; input >> _obj_DestPercentRMValue; input.ignore(1000, '\n');
	_obj.DestPercentRMValue = (float)_obj_DestPercentRMValue;
	// StructureBonus
	float _obj_StructureBonus; input >> _obj_StructureBonus; input.ignore(1000, '\n');
	_obj.StructureBonus = (float)_obj_StructureBonus;
	// MonsterBonus
	float _obj_MonsterBonus; input >> _obj_MonsterBonus; input.ignore(1000, '\n');
	_obj.MonsterBonus = (float)_obj_MonsterBonus;
	// CreepBonus
	float _obj_CreepBonus; input >> _obj_CreepBonus; input.ignore(1000, '\n');
	_obj.CreepBonus = (float)_obj_CreepBonus;
	return _obj;
}

File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientJava/src/StateAlterationModelView.java
import java.lang.*;

	public class StateAlterationModelView
	{

	
		public StateAlterationType Type;	
		public float BaseDuration;	
		public bool DashGoThroughWall;	
		public DashDirectionType DashDirectionType;	
		public float FlatValue;	
		public float SourcePercentADValue;	
		public float SourcePercentHPValue;	
		public float SourcePercentMaxHPValue;	
		public float SourcePercentArmorValue;	
		public float SourcePercentAPValue;	
		public float SourcePercentRMValue;	
		public float DestPercentADValue;	
		public float DestPercentHPValue;	
		public float DestPercentMaxHPValue;	
		public float DestPercentArmorValue;	
		public float DestPercentAPValue;	
		public float DestPercentRMValue;	
		public float StructureBonus;	
		public float MonsterBonus;	
		public float CreepBonus;	
		public static StateAlterationModelView Deserialize(BufferedReader input) {
		try {
			StateAlterationModelView _obj =  new StateAlterationModelView();
			// Type
			int _obj_Type = Integer.Parse(input.readLine());
			_obj.Type = StateAlterationType.fromValue(_obj_Type);
			// BaseDuration
			float _obj_BaseDuration = Float.Parse(input.readLine());
			_obj.BaseDuration = _obj_BaseDuration;
			// DashGoThroughWall
			boolean _obj_DashGoThroughWall = Integer.valueof(input.readLine()) == 0 ? false : true;
			_obj.DashGoThroughWall = _obj_DashGoThroughWall;
			// DashDirectionType
			int _obj_DashDirectionType = Integer.Parse(input.readLine());
			_obj.DashDirectionType = DashDirectionType.fromValue(_obj_DashDirectionType);
			// FlatValue
			float _obj_FlatValue = Float.Parse(input.readLine());
			_obj.FlatValue = _obj_FlatValue;
			// SourcePercentADValue
			float _obj_SourcePercentADValue = Float.Parse(input.readLine());
			_obj.SourcePercentADValue = _obj_SourcePercentADValue;
			// SourcePercentHPValue
			float _obj_SourcePercentHPValue = Float.Parse(input.readLine());
			_obj.SourcePercentHPValue = _obj_SourcePercentHPValue;
			// SourcePercentMaxHPValue
			float _obj_SourcePercentMaxHPValue = Float.Parse(input.readLine());
			_obj.SourcePercentMaxHPValue = _obj_SourcePercentMaxHPValue;
			// SourcePercentArmorValue
			float _obj_SourcePercentArmorValue = Float.Parse(input.readLine());
			_obj.SourcePercentArmorValue = _obj_SourcePercentArmorValue;
			// SourcePercentAPValue
			float _obj_SourcePercentAPValue = Float.Parse(input.readLine());
			_obj.SourcePercentAPValue = _obj_SourcePercentAPValue;
			// SourcePercentRMValue
			float _obj_SourcePercentRMValue = Float.Parse(input.readLine());
			_obj.SourcePercentRMValue = _obj_SourcePercentRMValue;
			// DestPercentADValue
			float _obj_DestPercentADValue = Float.Parse(input.readLine());
			_obj.DestPercentADValue = _obj_DestPercentADValue;
			// DestPercentHPValue
			float _obj_DestPercentHPValue = Float.Parse(input.readLine());
			_obj.DestPercentHPValue = _obj_DestPercentHPValue;
			// DestPercentMaxHPValue
			float _obj_DestPercentMaxHPValue = Float.Parse(input.readLine());
			_obj.DestPercentMaxHPValue = _obj_DestPercentMaxHPValue;
			// DestPercentArmorValue
			float _obj_DestPercentArmorValue = Float.Parse(input.readLine());
			_obj.DestPercentArmorValue = _obj_DestPercentArmorValue;
			// DestPercentAPValue
			float _obj_DestPercentAPValue = Float.Parse(input.readLine());
			_obj.DestPercentAPValue = _obj_DestPercentAPValue;
			// DestPercentRMValue
			float _obj_DestPercentRMValue = Float.Parse(input.readLine());
			_obj.DestPercentRMValue = _obj_DestPercentRMValue;
			// StructureBonus
			float _obj_StructureBonus = Float.Parse(input.readLine());
			_obj.StructureBonus = _obj_StructureBonus;
			// MonsterBonus
			float _obj_MonsterBonus = Float.Parse(input.readLine());
			_obj.MonsterBonus = _obj_MonsterBonus;
			// CreepBonus
			float _obj_CreepBonus = Float.Parse(input.readLine());
			_obj.CreepBonus = _obj_CreepBonus;
			} catch (UnsupportedEncodingExceptio e) { 
			} catch (IOException e) { }
			return _obj;
		}

		public void serialize(OutputStreamWriter output) {
			try {
			// Type
			output.append(((Integer)(this.Type.getValue())).toString() + "\n");
			// BaseDuration
			output.WriteLine(((Float)this.BaseDuration).toString() + "\n");
			// DashGoThroughWall
			output.append((this.DashGoThroughWall ? 1 : 0) + "\n");
			// DashDirectionType
			output.append(((Integer)(this.DashDirectionType.getValue())).toString() + "\n");
			// FlatValue
			output.WriteLine(((Float)this.FlatValue).toString() + "\n");
			// SourcePercentADValue
			output.WriteLine(((Float)this.SourcePercentADValue).toString() + "\n");
			// SourcePercentHPValue
			output.WriteLine(((Float)this.SourcePercentHPValue).toString() + "\n");
			// SourcePercentMaxHPValue
			output.WriteLine(((Float)this.SourcePercentMaxHPValue).toString() + "\n");
			// SourcePercentArmorValue
			output.WriteLine(((Float)this.SourcePercentArmorValue).toString() + "\n");
			// SourcePercentAPValue
			output.WriteLine(((Float)this.SourcePercentAPValue).toString() + "\n");
			// SourcePercentRMValue
			output.WriteLine(((Float)this.SourcePercentRMValue).toString() + "\n");
			// DestPercentADValue
			output.WriteLine(((Float)this.DestPercentADValue).toString() + "\n");
			// DestPercentHPValue
			output.WriteLine(((Float)this.DestPercentHPValue).toString() + "\n");
			// DestPercentMaxHPValue
			output.WriteLine(((Float)this.DestPercentMaxHPValue).toString() + "\n");
			// DestPercentArmorValue
			output.WriteLine(((Float)this.DestPercentArmorValue).toString() + "\n");
			// DestPercentAPValue
			output.WriteLine(((Float)this.DestPercentAPValue).toString() + "\n");
			// DestPercentRMValue
			output.WriteLine(((Float)this.DestPercentRMValue).toString() + "\n");
			// StructureBonus
			output.WriteLine(((Float)this.StructureBonus).toString() + "\n");
			// MonsterBonus
			output.WriteLine(((Float)this.MonsterBonus).toString() + "\n");
			// CreepBonus
			output.WriteLine(((Float)this.CreepBonus).toString() + "\n");
			} catch (UnsupportedEncodingExceptio e) { 
			} catch (IOException e) { }
		}

	}
}
